plugins/filter/filters.py

```diff
@@ -228,23 +228,38 @@ def listener_properties(self, listeners_dict, default_ssl_enabled,
 
             if self.normalize_sasl_protocol(listeners_dict[listener].get('sasl_protocol', default_sasl_protocol)) == 'OAUTHBEARER' and \
                     oauth_enabled and rbac_enabled:
-                final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] =\
-                    'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required publicKeyPath=\"' + \
-                    oauth_pem_path + '\" unsecuredLoginStringClaim_sub="thePrincipalName";'
                 final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.server.callback.handler.class'] =\
                     'io.confluent.kafka.server.plugins.auth.token.CompositeBearerValidatorCallbackHandler'
                 final_dict['listener.name.' + listener_name + '.sasl.oauthbearer.jwks.endpoint.url'] = oauth_jwks_uri
                 final_dict['listener.name.' + listener_name + '.sasl.oauthbearer.sub.claim.name'] = oauth_sub_claim
 
             if self.normalize_sasl_protocol(listeners_dict[listener].get('sasl_protocol', default_sasl_protocol)) == 'OAUTHBEARER' \
                     and oauth_enabled and not rbac_enabled:
-                final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] =\
-                    'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required unsecuredLoginStringClaim_sub="thePrincipalName";'
                 final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.server.callback.handler.class'] =\
                     'org.apache.kafka.common.security.oauthbearer.secured.OAuthBearerValidatorCallbackHandler'
                 final_dict['listener.name.' + listener_name + '.sasl.oauthbearer.jwks.endpoint.url'] = oauth_jwks_uri
                 final_dict['listener.name.' + listener_name + '.sasl.oauthbearer.sub.claim.name'] = oauth_sub_claim
 
+            if self.normalize_sasl_protocol(listeners_dict[listener].get('sasl_protocol', default_sasl_protocol)) == 'OAUTHBEARER' and \
+                    oauth_enabled:
+                if rbac_enabled and idp_self_signed:
+                    final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] =\
+                        'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required publicKeyPath=\"' + \
+                        oauth_pem_path + '\" ssl.truststore.location=\"' + kafka_broker_truststore_path + \
+                        '\" ssl.truststore.password=\"' + kafka_broker_truststore_storepass + '\" unsecuredLoginStringClaim_sub="thePrincipalName";'
+                if (rbac_enabled and (not idp_self_signed)):
+                    final_dict['listener.name.' + listener_name + '.oauthbearer.sasl.jaas.config'] =\
+                        'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required publicKeyPath=\"' + \
+                        oauth_pem_path + '\" unsecuredLoginStringClaim_sub="thePrincipalName";'
```

Comment:
since the name of this variable in playbooks has been changed should we also change this one

Reply:
this name is the python argument name, it is independent of the name we use in our playbooks

Reply:
Yes i get that it is independent of that variable. But the reason for maybe changing this would be just to make it clear while reading the code that this pem path is for certificate. (as even key files can be of pem format)

Reply:
agreed, updated it